src/utils/fonts.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Internal dependencies
+ */
+import { getFontFamilies } from '../resolvers';
+import { Font } from '../lib/lib-font/lib-font.browser';
+
+/**
+ * Fetch a file from a URL and return it as an ArrayBuffer.
+ *
+ * @param {string} url The URL of the file to fetch.
+ * @return {Promise<ArrayBuffer>} The file as an ArrayBuffer.
+ */
+async function fetchFileAsArrayBuffer( url ) {
+	try {
+		const response = await fetch( url );
+		if ( ! response.ok ) {
+			throw new Error( 'Network response was not ok.' );
+		}
+		const arrayBuffer = await response.arrayBuffer();
+		return arrayBuffer;
+	} catch ( error ) {
+		throw new Error(
+			`Failed to fetch file from the URL ${ url }. ${ error }`
+		);
+	}
+}
+
+/**
+ * Retrieves the licensing information of a font file given its URL.
+ *
+ * This function fetches the file as an ArrayBuffer, initializes a font object, and extracts licensing details from the font's OpenType tables.
+ *
+ * @param {string} url - The URL pointing directly to the font file. The URL should be a direct link to the file and publicly accessible.
+ * @return {Promise<Object|null>} A promise that resolves to an object containing the font's licensing details if successful. return null if the font license cannot be fetched or an error occurs.
+ *
+ * The returned object includes the following properties (if available in the font's OpenType tables):
+ *   - fontName: The full font name.
+ *   - copyright: Copyright notice.
+ *   - source: Unique identifier for the font's source.
+ *   - license: License description.
+ *   - licenseURL: URL to the full license text.
+ */
+async function getFontFileLicenseFromUrl( url ) {
+	try {
+		const buffer = await fetchFileAsArrayBuffer( url );
+		const fontObj = new Font( 'Uploaded Font' );
+		fontObj.fromDataBuffer( buffer, url );
+		// Assuming that fromDataBuffer triggers onload event and returning a Promise
+		const onloadEvent = await new Promise(
+			( resolve ) => ( fontObj.onload = resolve )
+		);
+		const font = onloadEvent.detail.font;
+		const { name: nameTable } = font.opentype.tables;
+		return {
+			fontName: nameTable.get( 16 ) || nameTable.get( 1 ),
+			copyright: nameTable.get( 0 ),
+			source: nameTable.get( 11 ),
+			license: nameTable.get( 13 ),
+			licenseURL: nameTable.get( 14 ),
+		};
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error( 'Failed to get font license for font file.', error );
```

Comment:
Remove console statement

```suggestion
		
```

Reply:
handled it here https://github.com/WordPress/create-block-theme/pull/649/commits/18839a3c37b6bff4cae725f81bd3b137c463c882